libs/@guardian/source-react-components-development-kitchen/src/footer-with-contents/FooterLinks.tsx

```diff
@@ -52,58 +64,82 @@ export interface FooterLinksProps {
 	/**
 	 * An array of links, specifying the link text and href, if the link is external, and any other HTML attributes for an anchor tag
 	 */
-	links?: FooterLink[];
+	links?: Array<FooterLink | FooterButton>;
 	/**
 	 * Force the links into a column layout below desktop, regardless of the amount of links
 	 */
 	forceColumns?: boolean;
 }
 
+const isFooterLink = (link: FooterLink | FooterButton): link is FooterLink => {
+	return 'href' in link;
+};
+
+const getLink = (link: FooterLink) => {
+	const { href, text, isExternal, ...linkAttrs } = link;
+	return (
+		<Link
+			cssOverrides={linkElementStyles}
+			href={href}
+			rel={isExternal ? 'noopener noreferrer' : ''}
+			{...linkAttrs}
+		>
+			{text}
+		</Link>
+	);
+};
+
+const getButtonLink = (link: FooterButton) => {
+	const { text, onClick, ...linkAttrs } = link;
+	return (
+		<ButtonLink
+			cssOverrides={linkElementStyles}
+			onClick={onClick}
+			{...linkAttrs}
+		>
+			{text}
+		</ButtonLink>
+	);
+};
+
 export const FooterLinks = ({
 	links = defaultGuardianLinks,
 	forceColumns = false,
 }: FooterLinksProps): EmotionJSX.Element => {
 	const useColumns = links.length > 3 || forceColumns;
+
 	return (
 		<div css={fullWidthContainer}>
 			<Hide from="tablet">
 				<ul css={getListStyles(useColumns)}>
-					{links.map(({ href, text, isExternal, ...linkAttrs }) => (
-						<li key={href} css={getItemStyles(useColumns)}>
-							<ThemeProvider theme={linkThemeBrand}>
-								<Link
-									cssOverrides={linkElementStyles}
-									href={href}
-									rel={isExternal ? 'noopener noreferrer' : ''}
-									{...linkAttrs}
-								>
-									{text}
-								</Link>
-							</ThemeProvider>
-						</li>
-					))}
+					{links.map((link, index) => {
+						{
+							return (
+								<li key={`link-${index}`} css={getItemStyles(useColumns)}>
+									<ThemeProvider theme={linkThemeBrand}>
+										{isFooterLink(link) ? getLink(link) : getButtonLink(link)}
```

Comment:
There’s a danger in using type predicates that do not have 100% coverage, so I would probably rely on TypeScript to do the narrowing for you inline:

```suggestion
										{'href' in link ? getLink(link) : getButtonLink(link)}
```

Alternatively a test proving that there is no way to construct an object that would incorrectly interpret a link or button would be very useful!